modules/library/xml/src/main/java/org/geotools/xml/resolver/SchemaCache.java

```diff
@@ -246,7 +246,9 @@ static void store(File file, byte[] bytes) {
             connection.setReadTimeout(downloadTimeout);
             connection.setUseCaches(false);
             connection.connect();
-            if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                if (hasRedirect(responseCode)) return followRedirect(connection, blockSize);
```

Comment:
While not very common, it's possible that the second response is another redirect. And so on.
Browsers follow them, but have some limits to avoid being caught in a redirect loop, e.g.:
https://www.liquidweb.com/kb/troubleshooting-too-many-redirects/

Reply:
fixed. Seemed reasonable to me to put a max number of redirect of 5.